django/db/models/base.py

```diff
@@ -75,6 +75,18 @@ def __new__(cls, name, bases, attrs, **kwargs):
         classcell = attrs.pop('__classcell__', None)
         if classcell is not None:
             new_attrs['__classcell__'] = classcell
+        # Fix related to bug #28198.
+        # Remove duplicate public attributes from parent classes
+        attr_per_class = {}
+        for parent in parents:
+            attr_per_parent = {}
+            for key in attrs.keys():
+                if hasattr(parent, key) and not key.startswith('__') and not hasattr(getattr(parent, key),
```

Comment:
This condition could use a comment with an explanation. `'__'` might be replaced with `LOOKUP_SEP` if that's what it's checking.

Reply:
isn't it checking for name mangling and/or dunder methods?